pyomo/contrib/parmest/parmest.py

```diff
@@ -1162,72 +1139,63 @@ def objective_at_theta(self, theta_values=None, initialize_parmest_model=False):
                 initialize_parmest_model=initialize_parmest_model,
             )
 
-        if len(self.estimator_theta_names) == 0:
+        if self.num_params == 0:
             pass  # skip assertion if model has no fitted parameters
         else:
             # create a local instance of the pyomo model to access model variables and parameters
             model_temp = self._create_parmest_model(0)
-            model_theta_list = self._expand_indexed_unknowns(model_temp)
-
-            # if self.estimator_theta_names is not the same as temp model_theta_list,
-            # create self.theta_names_updated
-            if set(self.estimator_theta_names) == set(model_theta_list) and len(
-                self.estimator_theta_names
-            ) == len(set(model_theta_list)):
-                pass
-            else:
-                self.theta_names_updated = model_theta_list
+            model_param_list = [c.name for c in _expand_indexed_unknowns(model_temp)]
 
         if theta_values is None:
-            all_thetas = {}  # dictionary to store fitted variables
-            # use appropriate theta names member
-            theta_names = model_theta_list
+            all_params = {}  # dictionary to store fitted variables
+            # use appropriate parameter names member
+            model_names = model_param_list
         else:
             assert isinstance(theta_values, pd.DataFrame)
             # for parallel code we need to use lists and dicts in the loop
-            theta_names = theta_values.columns
-            # # check if theta_names are in model
-            for theta in list(theta_names):
-                theta_temp = theta.replace("'", "")  # cleaning quotes from theta_names
-                assert theta_temp in [
-                    t.replace("'", "") for t in model_theta_list
-                ], "Theta name {} in 'theta_values' not in 'theta_names' {}".format(
-                    theta_temp, model_theta_list
+            model_names = theta_values.columns
+            # # check if model_names are in model
+            for param in list(model_names):
+                param_temp = param.replace("'", "")  # cleaning quotes from model_names
+                assert param_temp in [
+                    t.replace("'", "") for t in model_param_list
+                ], "Theta name {} in 'theta_values' not in parameter names {}".format(
```

Comment:
I think this might have been missed because it looks like you were trying to move away from using `theta`?
```suggestion
                ], "Param name {} in 'model_param_list' not in parameter names {}".format(
```